### PR TITLE
TST: Use indices fixture in tests/indexes/test_setops.py

### PR DESCRIPTION
**Pull request by SaturnFromTitan**
part of #30914
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

### CODE REVIEW

File: pandas/tests/indexes/test_setops.py [hunk 1]
```diff
@@ -39,14 +28,14 @@ def test_union_same_types(indices):
     assert idx1.union(idx2).dtype == idx1.dtype
 
 
-def test_union_different_types(index_pair):
+def test_union_different_types(indices, indices2):
```
Comment (simonjayhawkins) [inline pandas/tests/indexes/test_setops.py]: for a future reader, could add a comment along the lines that the test runs for all permutations of index types, but that this test only considers combinations.

**Review by simonjayhawkins — APPROVED** [review]
Thanks @SaturnFromTitan lgtm

**Comment by jbrockmendel** [comment]
LGTM

File: pandas/conftest.py [hunk 2]
```diff
@@ -404,6 +404,10 @@ def indices(request):
     return indices_dict[request.param].copy()
 
 
+# Needed to generate cartesian product of indices
```
Comment (jreback) [inline pandas/conftest.py]: this is a non-obvious name, call this index2_fixture

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/conftest.py (hunk 2 above)
Comment (SaturnFromTitan) [reply pandas/conftest.py]: Went with `index_fixture2` to be consistent with [nulls_fixture2](https://github.com/pandas-dev/pandas/blob/master/pandas/conftest.py#L267) and [unique_nulls_fixture2](https://github.com/pandas-dev/pandas/blob/master/pandas/conftest.py#L279)

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @SaturnFromTitan

**Pull request merged**